src/main/java/uk/gov/justice/digital/delius/service/NotificationService.java

```diff
@@ -33,10 +42,16 @@ public NotificationResponse notifyContact(final String crn, final Long sentenceI
 
         assertAppointmentType(request.getContactType());
 
-        final var newContact = makeNewContact(crn, sentenceId, request);
-        final var contactDto = deliusApiClient.createNewContact(newContact);
-
-        return NotificationResponse.builder().contactId(contactDto.getId()).build();
+        return matchExistingContact(crn, request)
+            .map(contactId -> {
+                log.debug(format("Notification already exists for CRN: %s, Sentence: %d, ContactType: %s and ContactDateTime: %s", crn, sentenceId, request.getContactType(), request.getContactDateTime()));
```

Comment:
This log message doesn't include everything used by the DB search. e.g. should it log NSI ID too?

Reply:
Yeah will log it and perhaps as a WARN to so we can see how often it arises